op-node/node/node.go

```diff
@@ -132,136 +152,214 @@ func New(ctx context.Context, cfg *config.Config, log log.Logger, appVersion str
 	return n, nil
 }
 
-func (n *OpNode) init(ctx context.Context, cfg *config.Config) error {
+type initializationOverrides struct {
+	L1Source   L1Source
+	Beacon     L1Beacon
+	RPCHandler *oprpc.Handler
+}
+
+// init progressively creates and sets up all the components of the OpNode
+// some later initialization steps depend on the node being partially initialized with other components,
+// so order is important to ensure that all resources are available when needed.
+func (n *OpNode) init(ctx context.Context, cfg *config.Config, overrides *initializationOverrides) error {
 	n.log.Info("Initializing rollup node", "version", n.appVersion)
-	n.initEventSystem()
-	if err := n.initL1BeaconAPI(ctx, cfg); err != nil {
-		return err
+	if overrides == nil {
+		overrides = &initializationOverrides{}
 	}
-	if err := n.initL1Source(ctx, cfg); err != nil {
-		return fmt.Errorf("failed to init L1 Source: %w", err)
+
+	eventSys, eventDrain, err := initEventSystem(n)
```

Comment:
This may also be a Go anti pattern but could we do:
```suggestion
    var err error
	n.eventSys, n.eventDrain, err = initEventSystem(n)
```
and save having to remember to manually assign `n.eventSys = eventSys` later (and similarly for all the other init calls)? I guess the compiler would complain if we didn't assign them since they'd then be unused currently...

Reply:
nope no compiler complaints, this was a great idea, tightened things up considerably.